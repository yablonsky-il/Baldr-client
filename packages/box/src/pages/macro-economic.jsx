import React from 'react';
import Container from '@material-ui/core/Container';

import { withCore } from 'core/hocs/with-core-component';
import { MacroEconomic as MacroEcomoicCore } from 'core/pages/macro-economic';

import { EconomicDataTabs } from '../components/economic-data-tabs/economic-data-tabs';
import { EconomicDataTable } from '../components/economic-data-table/economic-data-table';
import { DatePicker } from '../components/date-picker/date-picker';

class MacroEconomicUI extends React.Component {
  render() {
    const {
      selectedDate,
      pathname,
      setDate,
  } = this.props;

  console.log(this.props, 'this. props from box');

    return (
      <div>
        <Container maxWidth="xl" className="p-0">
        <div>
          <EconomicDataTabs />
          <DatePicker
            autoOk
            className="mt-4"
            inputVariant="outlined"
            label="With keyboard"
            value={selectedDate}
            onChange={setDate}
          />
        </div>
        <EconomicDataTable
          pathname={pathname}
        />
      </Container>
      </div>
    );
  }
}

export const MacroEconomic = withCore(MacroEcomoicCore, MacroEconomicUI);
