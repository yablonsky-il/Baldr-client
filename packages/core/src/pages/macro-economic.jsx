import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import * as R from 'ramda';

import {
  fetchEconomicData as fetchEconomicDataAction,
  clearEconomicData as clearEconomicDataAction,
} from '../actions/economic-data';
import { getDate, isEmptyOrNil } from '../helpers/util';

const getParam = pathname => R.last(R.split('/')(pathname));

export class MacroEconomicUI extends PureComponent {
  state = {
    selectedDate: new Date(),
    pathname: null,
  };

  static getDerivedStateFromProps(props, state) {
    const { pathname: prevPathname } = state;
    const {
      fetchEconomicData,
      location: { pathname: currentPathname },
    } = props;
    const indicator = getParam(currentPathname);

    if (!R.equals(currentPathname, prevPathname) && !isEmptyOrNil(indicator)) {
      fetchEconomicData({
        date: getDate(state.selectedDate),
        indicator,
      });

      return { ...state, pathname: currentPathname };
    }

    return null;
  }

  componentDidMount() {
    const { selectedDate } = this.state;
    const {
      fetchEconomicData,
      location: { pathname: currentPathname },
    } = this.props;

    const indicator = getParam(currentPathname);

    if (!isEmptyOrNil(indicator)) {
      fetchEconomicData({
        date: getDate(selectedDate),
        indicator,
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    const { selectedDate: prevDate } = prevState;
    const { selectedDate } = this.state;

    if (!R.equals(selectedDate, prevDate)) {
      const {
        fetchEconomicData,
        location: { pathname },
      } = this.props;

      fetchEconomicData({
        date: getDate(selectedDate),
        indicator: getParam(pathname),
      });
    }
  }

  componentWillUnmount() {
    const { clearEconomicData } = this.props;

    clearEconomicData();
  }

  setDate = newDate => this.setState(prevState => ({
    ...prevState,
    selectedDate: newDate,
  }))

  render() {
    const { selectedDate } = this.state;
    const {
      children,
      isInProgress,
      location: { pathname },
      economicData,
    } = this.props;

    return children({
      pathname,
      selectedDate,
      isInProgress,
      economicData,
      setDate: this.setDate,
    });
  }
}

const mapStateToProps = ({
  macroEconomic: {
    isInProgress,
    economicData,
  },
}) => ({
  isInProgress,
  economicData,
});

const mapDispatchToProps = {
  fetchEconomicData: fetchEconomicDataAction,
  clearEconomicData: clearEconomicDataAction,
};

export const MacroEconomic = withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(MacroEconomicUI));
