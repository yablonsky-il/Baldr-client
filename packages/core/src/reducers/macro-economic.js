import { handleActions } from 'redux-actions';

import {
  fetchEconomicData,
  setEconomicData,
  setFetchStatus,
  clearEconomicData,
} from '../actions/economic-data';

const getInitialState = () => ({
  isInProgress: false,
  isFetched: false,
  status: { message: null, value: null },
  economicData: {
    date: null,
    data: null,
    indicator: null,
  },
});

export const macroEconomic = handleActions({
  [fetchEconomicData]: state => ({
    ...state,
    isInProgress: true,
  }),
  [setEconomicData]: (state, { payload }) => ({ // payload: { date: string, data: object, indicator: string }
    ...state,
    isInProgress: false,
    isFetched: true,
    economicData: { ...payload },
  }),
  [setFetchStatus]: (state, { payload }) => ({ // payload: string
    ...state,
    status: { ...payload },
  }),
  [clearEconomicData]: getInitialState,
}, getInitialState());
