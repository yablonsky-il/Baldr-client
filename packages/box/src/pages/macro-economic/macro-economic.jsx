import React from 'react';
import Container from '@material-ui/core/Container';
import PropTypes from 'prop-types';

import { withCore } from 'core/hocs/with-core-component';
import { MacroEconomic as MacroEcomoicCore } from 'core/pages/macro-economic';

import { EconomicDataTabs } from '../../components/economic-data-tabs/economic-data-tabs';
import { EconomicDataTable } from '../../components/economic-data-table/economic-data-table';
import { DatePicker } from '../../components/date-picker/date-picker';

export const MacroEconomicUI = ({
  isInProgress,
  economicData,
  selectedDate,
  pathname,
  setDate,
}) => (
  <Container maxWidth="xl" className="p-0">
    <div>
      <EconomicDataTabs isInProgress={isInProgress} />
      <DatePicker
        autoOk
        className="mt-4"
        inputVariant="outlined"
        label="With keyboard"
        value={selectedDate}
        onChange={setDate}
      />
    </div>
    <EconomicDataTable
      pathname={pathname}
      economicData={economicData}
    />
  </Container>
);

MacroEconomicUI.propTypes = {
  isInProgress: PropTypes.bool.isRequired,
  economicData: PropTypes.shape({}).isRequired,
  selectedDate: PropTypes.shape({}).isRequired,
  pathname: PropTypes.string.isRequired,
  setDate: PropTypes.func.isRequired,
};

const MacroEconomic = withCore(MacroEcomoicCore, MacroEconomicUI);

export default MacroEconomic;
