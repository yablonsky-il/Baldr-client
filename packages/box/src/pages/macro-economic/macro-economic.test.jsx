import { MacroEconomicUI } from './macro-economic';

const props = {
  isInProgress: false,
  economicData: { date: null, data: null },
  selectedDate: null,
  pathname: 'en/home',
  setDate: null,
};

describe('<HomeUI />', () => {
  const wrapper = shallow(<MacroEconomicUI {...props} />);

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
