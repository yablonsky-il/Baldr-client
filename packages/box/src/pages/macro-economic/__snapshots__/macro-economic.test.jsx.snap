// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeUI /> should match snapshot 1`] = `
<WithStyles(ForwardRef(Container))
  className="p-0"
  maxWidth="xl"
>
  <div>
    <withRouter(EconomicDataTabsUI)
      isInProgress={false}
    />
    <DatePicker
      autoOk={true}
      className="mt-4"
      inputVariant="outlined"
      label="With keyboard"
      onChange={null}
      value={null}
    />
  </div>
  <EconomicDataTable
    economicData={
      Object {
        "data": null,
        "date": null,
      }
    }
    pathname="en/home"
  />
</WithStyles(ForwardRef(Container))>
`;
